sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -114,6 +114,113 @@ object JavaCode {
   }
 }
 
+/**
+ * A trait representing a block of java code.
+ */
+trait Block extends JavaCode {
+
+  // The expressions to be evaluated inside this block.
+  def exprValues: Seq[ExprValue]
```

Comment:
These are the inputs right?

Reply:
They are `ExprValues` in the inputs. Inputs may contain other than `ExprValues`, e.g. other blocks, literals...

Reply:
Not sure the order among those `ExprValue`s will be useful or not. Sounds a `Set` is good.

Reply:
yes, I think it is not and we are keeping here the order of the first occurrence (so for instance if we want to check which is the last reference we can't really know). I think a `Set` would be better since it enforces what we are keeping here, it avoids all the distinct (which we may forget to add somewhere getting a bad result...)

Reply:
One issue I just found is, I can use the `ExprValues` order to do test easily.

Reply:
isn't it enough to compare `exprValues` with a `Set` containing the expected  elements instead of comparing them one by one?

Reply:
Yes, you're right. Sorry, just out of focus.